R/kelp.R

```diff
@@ -0,0 +1,110 @@
+#' Create a kelp object for interacting with SeaweedFS.
+#'
+#' This is a slightly higher level abstraction than [seaweed_master()] and
+#' [seaweed_volume()].
+#'
+#' @keywords export
+#' @importFrom R6 R6Class
+kelp <- R6::R6Class(
+  "kelp",
+  cloneable = FALSE,
+
+  public = list(
+    #' @field master A [seaweed_master] object for sending requests
+    master = NULL,
+
+    #' @description
+    #' Create client object for sending http requests to seaweed
+    #'
+    #' @param seaweed_url Root URL of Seaweed
+    #'
+    #' @return A new `kelp` object
+    initialize = function(seaweed_url) {
+      self$master <- seaweed_master$new(seaweed_url)
+    },
+
+    #' @description
+    #' Upload file to SeaweedFS
+    #'
+    #' @param path Path to file to be uploaded
+    #' @param collection Collection name, acts as a namespace for files.
+    #'
+    #' @return The uploaded file ID.
+    upload_file = function(path, collection = NULL) {
```

Comment:
Interface question: do we want upload/download or (put|set)/get here for the interface?

Reply:
I went with `upload` on upload because it matches terminology SeaweedFS uses (and then download as the inverse of that). But really don't feel too strongly